File: frontend/src/services/health.ts
import { request } from "@/utils/request";

export interface HealthResponse {
  status: "ok";
  service: string;
  version: string;
  environment: string;
}

export const healthApi = {
  getLive() {
    return request.get<HealthResponse>("/health/live");
  },
};

File: backend/src/techbrain/schemas/health.py
"""Health endpoint schemas."""

from typing import Literal

from pydantic import BaseModel

from techbrain.core.config import Environment


class HealthResponse(BaseModel):
    """API process health."""

    status: Literal["ok", "error"]
    service: str
    version: str
    environment: Environment


class ReadinessCheck(BaseModel):
    """One readiness dependency check."""

    name: str
    status: Literal["ok", "error"]
    message: str | None = None


class ReadinessResponse(HealthResponse):
    """API readiness including dependency checks."""

    checks: list[ReadinessCheck]
